plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlab.ts

```diff
@@ -174,20 +174,30 @@ export function createPublishGitlabAction(options: {
         [tokenType]: token,
       });
 
-      let { id: targetNamespace } = (await client.Namespaces.show(owner)) as {
-        id: number;
-      };
+      let targetNamespaceId;
+
+      try {
+        const namespaceResponse = (await client.Namespaces.show(owner)) as {
+          id: number;
+        };
+
+        targetNamespaceId = namespaceResponse.id;
+      } catch (e) {
+        throw new InputError(
+          `The namespace ${owner} is not found or the user doesn't have permissions to access it`,
```

Comment:
this logic returns the message for any error, instead of only 404 errors. It would be better to only catch 404 errors

Reply:
Totally makes sense! Done.